From 7a76b46592b62012fe9b8308b59568e3a15f0329 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Fri, 8 Dec 2017 15:35:36 +0800
Subject: [PATCH 040/129] armv8: ls1012ardb: add more board version information

Add LS1012ARDB RevC/RevC1/RevC2/RevD/RevE information and
detect it when u-boot starts up.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
Reviewed-by: York Sun <york.sun@nxp.com>
---
 board/freescale/ls1012ardb/ls1012ardb.c |   26 +++++++++++++++++++++++---
 include/configs/ls1012ardb.h            |    5 +++++
 2 files changed, 28 insertions(+), 3 deletions(-)

diff --git a/board/freescale/ls1012ardb/ls1012ardb.c b/board/freescale/ls1012ardb/ls1012ardb.c
index 2f1d637..421b0a7 100644
--- a/board/freescale/ls1012ardb/ls1012ardb.c
+++ b/board/freescale/ls1012ardb/ls1012ardb.c
@@ -41,12 +41,32 @@ int checkboard(void)
 	}
 
 	puts("Version");
-	if ((in1 & SW_REV_MASK) == SW_REV_A)
+	switch (in1 & SW_REV_MASK) {
+	case SW_REV_A:
 		puts(": RevA");
-	else if ((in1 & SW_REV_MASK) == SW_REV_B)
+		break;
+	case SW_REV_B:
 		puts(": RevB");
-	else
+		break;
+	case SW_REV_C:
+		puts(": RevC");
+		break;
+	case SW_REV_C1:
+		puts(": RevC1");
+		break;
+	case SW_REV_C2:
+		puts(": RevC2");
+		break;
+	case SW_REV_D:
+		puts(": RevD");
+		break;
+	case SW_REV_E:
+		puts(": RevE");
+		break;
+	default:
 		puts(": unknown");
+		break;
+	}
 
 	printf(", boot from QSPI");
 	if ((in1 & SW_BOOT_MASK) == SW_BOOT_EMU)
diff --git a/include/configs/ls1012ardb.h b/include/configs/ls1012ardb.h
index 7065f83..9dcedaf 100644
--- a/include/configs/ls1012ardb.h
+++ b/include/configs/ls1012ardb.h
@@ -34,6 +34,11 @@
 #define SW_REV_MASK		0xF8
 #define SW_REV_A		0xF8
 #define SW_REV_B		0xF0
+#define SW_REV_C		0xE8
+#define SW_REV_C1		0xE0
+#define SW_REV_C2		0xD8
+#define SW_REV_D		0xD0
+#define SW_REV_E		0xC8
 
 /*  MMC  */
 #ifdef CONFIG_MMC
-- 
1.7.1

